# Projeto Top10 diretores com o maior nº de obras na Netflix


Um projeto bem simples utilizado apenas para estudo, focado em manipulação e análise de dados.



# Início

In [ ]:
import pandas as pd 
import seaborn as srn
import matplotlib.pyplot as plt
 

In [ ]:
df=pd.read_csv('netflix1.csv')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8790 non-null   object
 1   type          8790 non-null   object
 2   title         8790 non-null   object
 3   director      8790 non-null   object
 4   country       8790 non-null   object
 5   date_added    8790 non-null   object
 6   release_year  8790 non-null   int64 
 7   rating        8790 non-null   object
 8   duration      8790 non-null   object
 9   listed_in     8790 non-null   object
dtypes: int64(1), object(9)
memory usage: 686.8+ KB


In [ ]:
df.head()

show_id     type                             title         director  \
0      s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1      s3  TV Show                         Ganglands  Julien Leclercq   
2      s6  TV Show                     Midnight Mass    Mike Flanagan   
3     s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4      s8    Movie                           Sankofa     Haile Gerima   

         country date_added  release_year rating  duration  \
0  United States  9/25/2021          2020  PG-13    90 min   
1         France  9/24/2021          2021  TV-MA  1 Season   
2  United States  9/24/2021          2021  TV-MA  1 Season   
3         Brazil  9/22/2021          2021  TV-PG    91 min   
4  United States  9/24/2021          1993  TV-MA   125 min   

                                           listed_in  
0                                      Documentaries  
1  Crime TV Shows, International TV Shows, TV Act...  
2                 TV Dramas, TV Horror, TV Mysteries  
3                 Children & Family Movies, Comedies  
4   Dramas, Independent Movies, International Movies

# CORREÇAÕ DE DATA (datetime)

In [ ]:
df['date_added']= pd.to_datetime(df['date_added'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   show_id       8790 non-null   object        
 1   type          8790 non-null   object        
 2   title         8790 non-null   object        
 3   director      8790 non-null   object        
 4   country       8790 non-null   object        
 5   date_added    8790 non-null   datetime64[ns]
 6   release_year  8790 non-null   int64         
 7   rating        8790 non-null   object        
 8   duration      8790 non-null   object        
 9   listed_in     8790 non-null   object        
dtypes: datetime64[ns](1), int64(1), object(8)
memory usage: 686.8+ KB


In [ ]:
df['year_added'] = df['date_added'].dt.year
df
#colocando a o ano de lançamento através de uma nova coluna chamada 'year_added'

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country date_added  release_year rating   duration  \
0     United States 2021-09-25          2020  PG-13     90 min   
1            France 2021-09-24          2021  TV-MA   1 Season   
2     United States 2021-09-24          2021  TV-MA   1 Season   
3            Brazil 2021-09-22          2021  TV-PG     91 min   
4     United States 2021-09-24          1993  TV-MA    125 min   
...             ...        ...           ...    ...        ...   
8785         Turkey 2017-01-17          2016  TV-PG  2 Seasons   
8786  United States 2018-09-13          2016  TV-Y7  3 Seasons   
8787       Pakistan 2016-12-15          2012  TV-PG   1 Season   
8788       Pakistan 2018-06-23          2016   TV-Y   1 Season   
8789       Pakistan 2018-06-07          2016  TV-Y7   1 Season   

                                              listed_in  year_added  
0                                         Documentaries        2021  
1     Crime TV Shows, International TV Shows, TV Act...        2021  
2                    TV Dramas, TV Horror, TV Mysteries        2021  
3                    Children & Family Movies, Comedies        2021  
4      Dramas, Independent Movies, International Movies        2021  
...                                                 ...         ...  
8785                  International TV Shows, TV Dramas        2017  
8786                                           Kids' TV        2018  
8787  International TV Shows, Romantic TV Shows, TV ...        2016  
8788                                           Kids' TV        2018  
8789                                           Kids' TV        2018  

[8790 rows x 11 columns]

#ANÁLISE EXPLORATÓRIA


In [ ]:
#identificando quantos títulos possuem em 'Movie" e em 'TV show'
types =df.groupby(['type',])['type'].count().reset_index(name='count')
types = types.set_index('type')
types

count
type          
Movie     6126
TV Show   2664

Aqui podemos concluir que temos 6126 filmes e 2664 programas de TV.


In [ ]:
#Agrupando diretores, tipo de progarma, e quantos 'shows'
show_director = df.groupby(['director','type'])['director'].count().reset_index(name= 'show_count')
show_director

director   type  show_count
0             A. L. Vijay  Movie           2
1            A. Raajdheep  Movie           1
2               A. Salaam  Movie           1
3         A.R. Murugadoss  Movie           2
4         Aadish Keluskar  Movie           1
...                   ...    ...         ...
4576          Çagan Irmak  Movie           1
4577     Ísold Uggadóttir  Movie           1
4578  Óskar Thór Axelsson  Movie           1
4579     Ömer Faruk Sorak  Movie           2
4580         Şenol Sönmez  Movie           2

[4581 rows x 3 columns]

In [ ]:
#top 10 melhores diretores
top10 = show_director.query(" show_count >=12")
top10

director     type  show_count
127      Alastair Fothergill  TV Show          14
690      Cathy Garcia-Molina    Movie          13
1800             Jay Chapman    Movie          12
1803               Jay Karas    Movie          14
2623            Marcus Raboy    Movie          15
2699         Martin Scorsese    Movie          12
3100               Not Given    Movie         173
3101               Not Given  TV Show        2415
3432           Rajiv Chilaka    Movie          19
3484  Raúl Campos, Jan Suter    Movie          18
4092             Suhas Kadav    Movie          16
4532         Youssef Chahine    Movie          12

Chegamos a conclusão dos 10 diretores com maior qunantidade de obras da netflix. "Not Given" são dados como não específicado.

